fn_cve_search/fn_cve_search/components/function_cve_browse.py

```diff
@@ -0,0 +1,118 @@
+        if isinstance(api_data, list):
+            raise NotImplementedError()
+        elif isinstance(api_data, dict):
+            for key_data, value_data in api_data.items():
+                tmp_browse_list.append({key_data: value_data})
+            return tmp_browse_list
+        else:
+            raise NotImplementedError()
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_cve_search", {})
+
+    @function("function_cve_browse")
+    def _function_cve_browse_function(self, event, *args, **kwargs):
+        """Function: A Function to Browse Common Vulnerability Exposures  Vendors and Product & Database information from https://cve.circl.lu Data Base."""
+        try:
+            # Get the function parameters:
+            cve_browse_data = kwargs.get("cve_browse_data")  # text
+            if cve_browse_data:
+                cve_browse_data = cve_browse_data.strip()
+
+            cve_browse_criteria = kwargs.get("cve_browse_criteria")  # text
+            if cve_browse_criteria:
+                cve_browse_criteria = cve_browse_criteria.strip()
+
+            cve_vendor = kwargs.get("cve_vendor")  # text
+            if cve_vendor:
+                cve_vendor = cve_vendor.strip()
+
+            # Variables to Store Parsed CVE API Data
+            result_data_dict = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_browse_data: %s", cve_browse_data)
+            log.info("cve_browse_criteria: %s", cve_browse_criteria)
+            log.info("cve_vendor: %s", cve_vendor)
+
+            yield StatusMessage("starting...")
+
+            if cve_browse_criteria.lower().find('browse') != -1:
+                browse_data = self._browse_cve_api(vendor_name=cve_vendor)
+            elif cve_browse_criteria.lower().find('db') != -1:
+                browse_data = self._cve_db_information()
+            else:
+                raise ValueError("CVE Browse Criteria is not recognized..!")
+
+            # Defining a key in result dictionary to store parsed data
+            result_data_dict['content'] = []
+
+            # Type Of the rest api call like browse,search,specific cve,last, db info
+            api_call_type = browse_data['api_call']
+
+            # Rest Api Response Data
+            browse_data_content = browse_data['content']
+
+            if api_call_type == 'browse':
+                result_data_dict['api_call'] = 'browse'
+                _browse_dict_list = self._parse_browse_results(api_data=browse_data_content)
+                result_data_dict['content'].extend(_browse_dict_list)
+            elif api_call_type == 'db':
+                result_data_dict['api_call'] = 'db'
+                result_data_dict['content'].append(browse_data_content)
+
+            log.debug("The Data Received from CVE: {}".format(result_data_dict))
+            yield StatusMessage("done...")
```

Comment:
A better message like: "List of Vendors received from CVE Database"

Reply:
ok